.gdbinit

```diff
@@ -3,7 +3,7 @@ file build/main.elf
 load
 
 # b main
-# b mc_entry
-# b runStateMachine
-b runLEDThread
+b mc_entry
+b runStateMachine
+b state_machine_thread.cpp:44
```

Comment:
like this is fine but it would make more sense to do it on the function name so that we don't like paste in a bunch of lines ahead of it and mess up this line number

Reply:
Line 44 is the switch statement in the while loop so I didn't have to step through the entire loop every time. I didn't intend to commit this, it just got included in with `git add .`

On this topic, I'm actually not sure how we should go about putting `.gdbinit` under source control. GDB has a Python API that we can use to automate certain things: https://interrupt.memfault.com/blog/automate-debugging-with-gdb-python-api. This is definitely something we should look into going into G6 @ren-andy @Nushaab-Syed. We could even use this to do unit testing.

Reply:
Writing unit tests using this could also be a good onboarding activity. It gets you familiar with the codebase and GDB.